tutorials/speaker_recognition/ASR_with_SpeakerDiarization.ipynb

```diff
@@ -0,0 +1,547 @@
+    "    spaces.append([idx_state, len(pred)-1])\n",
+    "# calibration offset for timestamps: 180 ms\n",
+    "offset = -0.18\n",
+    "\n",
+    "# split the transcript into words\n",
+    "words = transcript.split()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Frame level stamps for non speech frames "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(non_speech)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "write to rttm type file for later use in extracting speaker labels"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "frame_offset=offset/time_stride\n",
+    "speech_labels=[]\n",
+    "uniq_id = os.path.basename(AUDIO_FILENAME).split('.')[0]\n",
+    "with open(uniq_id+'.rttm','w') as f:\n",
+    "    for idx in range(len(non_speech)-1):\n",
+    "        start = (non_speech[idx][1]+frame_offset)*time_stride\n",
+    "        end = (non_speech[idx+1][0]+frame_offset)*time_stride\n",
+    "        f.write(\"SPEAKER {} 1 {:.3f} {:.3f} <NA> <NA> speech <NA>\\n\".format(uniq_id,start,end-start))\n",
+    "        speech_labels.append(\"{:.3f} {:.3f} speech\".format(start,end))\n",
+    "    if non_speech[-1][1] < len(probs):\n",
+    "        start = (non_speech[-1][1]+frame_offset)*time_stride\n",
+    "        end = (len(probs)+frame_offset)*time_stride\n",
+    "        f.write(\"SPEAKER {} 1 {:.3f} {:.3f} <NA> <NA> speech <NA>\\n\".format(uniq_id,start,end-start))\n",
+    "        speech_labels.append(\"{:.3f} {:.3f} speech\".format(start,end))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Time stamps for speech frames"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(speech_labels)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "COLORS=\"b g c m y\".split()\n",
+    "def get_color(signal,speech_labels,sample_rate=16000):\n",
+    "    c=np.array(['k']*len(signal))\n",
+    "    for time_stamp in speech_labels:\n",
+    "        start,end,label=time_stamp.split()\n",
+    "        start,end = int(float(start)*16000),int(float(end)*16000),\n",
+    "        if label == \"speech\":\n",
+    "            code = 'red'\n",
+    "        else:\n",
+    "            code = COLORS[int(label.split('_')[-1])]\n",
+    "        c[start:end]=code\n",
+    "    \n",
+    "    return c "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With voice activity time stamps extracted from CTC outputs, Here we show the Voicy Activity signal in <span style=\"color:red\">**red**</span> color while background speech in **black** color"
```

Comment:
With voice activity time stamps extracted from CTC outputs, here we show the Voice Activity signal in **red** color and background speech in black color
Corrected for typos ^^